File: Semestru II/Tehnologii_Web/Laborator/Lab7/xml-dom/transformari/dotnet/xml-transformer.cs
/*
   Un program C# utilizat la transformarea, 
   via o foaie XSL, a documentelor XML 
   (varianta pentru .NET Framework 2.0) 
 
   A se compila cu: csc xml-transformer.cs
   A se executa cu: xml-transformer <fisier.xml> <fisier.xsl>
   
   Autor: Sabin Buraga -- http://www.purl.org/net/busaco
   Portiuni din acest program apartin Microsoft
   Ultima actualizare: 18 octombrie 2007
*/
using System;
using System.IO;
using System.Xml;
using System.Xml.Xsl;
using System.Xml.XPath;

namespace csbXMLTools
{
  class XMLTransformer // clasa de transformare a documentelor XML
  {
  		
    // 'programul' principal
    public static void Main(string[] args)
    {
      // n-au fost furnizate suficiente argumente
      if ( args.Length < 2 )
      {
        Console.WriteLine("Sintaxa: xml-transformer <fisier.xml> <fisier.xsl>\n");
        Console.WriteLine("Transforma documente XML folosind o foaie XSL.");
        return;
      }	
      
      // instanta a clasei de transformare optimizata
      XslCompiledTransform xslt = null;
      
      // incercam sa procesam documentul XML
      try 
      {
      // instantiem cititorul de continut XML	
      xslt = new XslCompiledTransform();
      
      // incarcam foaia de stiluri XSL
      xslt.Load(args[1]);
	
      // realizam transformarea
      xslt.Transform(new XPathDocument(args[0]), 
                     new XmlTextWriter(Console.Out));

      } // final de "try"
      catch ( Exception e ) 
      {
      	// a aparut o exceptie
      	Console.WriteLine("Exceptie: " + e.Message);
      	return;
      }
      // gata!
    }    
    
  } // final de clasa
} // final de spatiu de nume  

File: Semestru II/Tehnologii_Web/Laborator/Lab7/xml-dom/transformari/java/XMLTransformer.java
/* Program Java care realizeaza transformari XSL

   A se compila cu: javac XMLTransformer.java
   A se executa cu: java XMLTransformer

   Autor: Sabin-Corneliu Buraga (2007) -- http://www.infoiasi.ro/~busaco/
   Ultima actualizare: 18 martie 2007
*/
import javax.xml.transform.*;
import javax.xml.transform.stream.*;

public class XMLTransformer {

	// 'programul' principal
	// args[0] reprezinta URI-ul documentului XML
	// args[1] reprezinta URI-ul documentului XSLT
	public static void main (String[] args) {

		if (args.length != 2) {
      System.err.println
        ("Realizeaza transformari XSL.\nSintaxa: XMLTransformer <document.xml> <foaie.xsl>");
      return;
    }

		// instantieri...
		TransformerFactory tr = TransformerFactory.newInstance();
		Transformer trans;

		try {
			// cream un transformator, folosind foaia de stiluri XSLT
			trans = tr.newTransformer (new StreamSource (args[1]));
			// cream sursa de date XML (aici, un fisier)
			StreamSource xml = new StreamSource (args[0]);
			// rezultatul transformarii e redirectionat spre iesirea standard
			StreamResult rez = new StreamResult (System.out);
			// realizam transformarea
			trans.transform (xml, rez);
		}
		// tratam exceptiile
		catch (TransformerConfigurationException e) {
			System.err.println ("Eroare la initializare.\n" + e);
		}
		catch (TransformerException e) {
			System.err.println ("Eroare la transformare.\n" + e);
		}
	}
}